doc/user_guide/interactions/parameters.rst

```diff
@@ -0,0 +1,485 @@
+dataset.
+
+Each selection type has attributes through which its behavior can be
+customized; for example we might wish for our brush to be tied only
+to the ``"x"`` encoding to emphasize that feature in the data.
+We can modify the brush definition, and leave the rest of the code unchanged:
+
+.. altair-plot::
+
+    brush = alt.selection_interval(encodings=['x'])
+
+    chart = alt.Chart(cars).mark_point().encode(
+        x='Horsepower:Q',
+        y='Miles_per_Gallon:Q',
+        color=alt.condition(brush, 'Origin:N', alt.value('lightgray'))
+    ).properties(
+        width=250,
+        height=250
+    ).add_params(
+        brush
+    )
+
+    chart | chart.encode(x='Acceleration:Q')
+
+As you might have noticed,
+the selected points are sometimes obscured by some of the unselected points.
+To bring the selected points to the foreground,
+we can change the order in which they are laid out via the following
+encoding: ``order=alt.condition(hover, alt.value(1), alt.value(0))``.
+You can see and example of this in the :ref:`gallery_selection_zorder` gallery example.
+
+Filtering Data
+^^^^^^^^^^^^^^
+
+Using a selection parameter to filter data works in much the same way
+as using it within ``condition``,
+For example, in ``transform_filter(brush)``,
+we are again using the selection parameter ``brush`` as a predicate.
+Data points which evaluate to ``True`` (i.e., data points which lie within the selection) are kept,
+and data points which evaluate to ``False`` are filtered out.
+
+It is not possible to both select and filter in the same chart,
+so typically this functionality will be used when at least two sub-charts are present.
+In the following example,
+we attach the selection parameter to the upper chart,
+and then filter data in the lower chart based on the selection in the upper chart.
+You can explore how the counts change in the bar chart
+depending on the size and position of the selection in the scatter plot.
+
+.. altair-plot::
+
+    brush = alt.selection_interval()
+
+    points = alt.Chart(cars).mark_point().encode(
+        x='Horsepower:Q',
+        y='Miles_per_Gallon:Q',
+        color='Origin:N'
+    ).add_params(
+        brush
+    )
+
+    bars = alt.Chart(cars).mark_bar().encode(
+        x='count()',
+        y='Origin:N',
+        color='Origin:N'
+    ).transform_filter(
+        brush
+    )
+
+    points & bars
+
+
+Selection Types
+~~~~~~~~~~~~~~~
+
+Now that we have seen the basics of how we can use a selection to interact with a chart,
+let's take a more systematic look at some of the types of selection parameters available in Altair.
+For simplicity, we'll use a common chart in all the following examples; a
+simple heat-map based on the ``cars`` dataset.
+For convenience, let's write a quick Python function that will take a selection
+object and create a chart with the color of the chart elements linked to this
+selection:
+
+.. altair-plot::
+    :output: none
+
+    def make_example(selector):
```

Comment:
I think this concisely reinforces the concept of a selection being a type of parameter
```suggestion
    def make_example(selector: alt.Parameter) -> alt.Chart:
```